### ISSUE TITLE
PipeReader.CompleteAsync no longer propagates exceptions to PipeWriter

### ISSUE THREAD

**Issue by AArnott** [dotnet/runtime#43640]
### Description

When `PipeReader.CompleteAsync(Exception)` is called with a non-null exception, I expect subsequent calls to `PipeWriter.WriteAsync` to throw that exception.

### Repro

The following pair of tests pass on System.IO.Pipelines 4.7.2. The latter test fails on System.IO.Pipelines 4.7.3.

```cs
    [Fact]
    public async Task PipeWriter_PropagatesExceptions()
    {
        var pipe = new Pipe();
        await pipe.Writer.CompleteAsync(new Exception());
        await Assert.ThrowsAsync<Exception>(async () => await pipe.Reader.ReadAsync(this.TimeoutToken));
    }

    [Fact]
    public async Task PipeReader_PropagatesExceptions()
    {
        var pipe = new Pipe();
        await pipe.Reader.CompleteAsync(new Exception());
        await Assert.ThrowsAsync<Exception>(async () => await pipe.Writer.WriteAsync(new byte[1], this.TimeoutToken));
    }
```

### Configuration

This fails on .NET Framework, .NET Core 2.1 and .NET Core 3.1, on all operating systems.

### Regression?

This regressed in System.IO.Pipelines 4.7.3. v4.7.2 did not have this bug.

**Comment by davidfowl** [comment]
😭 Now we'll need to patch the patch.

**Comment by danmosemsft** [comment]
@BrennanConroy this is set 5.0.x -- do you plan to patch in 5.0.1? Code complete is 11/25.

**Comment by BrennanConroy** [comment]
Yes, already have approval https://github.com/dotnet/runtime/pull/43903

**Comment by AArnott** [comment]
So... we're *not* going to patch 4.7.3 in a 4.7.4 release?

**Comment by BrennanConroy** [comment]
https://github.com/dotnet/corefx/pull/43002 is for 3.1 (4.7.4)

**Comment by BrennanConroy** [comment]
All PRs merged.

**Comment by AArnott** [comment]
I haven't seen this fix shipped for either the 4.x or 5.x trains yet. When will it be pushed to nuget.org?

**Comment by BrennanConroy** [comment]
For 5.0, ~1 hour ago https://www.nuget.org/packages/System.IO.Pipelines/5.0.1
For 3.1, sometime in January.
